### ISSUE TITLE
PANIC: zfs: accessing past end of object 309/355c (size=1536 access=1018+1026)

### ISSUE THREAD

**Issue by pengc99** [openzfs/zfs#10904]
### System information
Distribution Name	| Debian
Distribution Version	| 10.5
Linux Kernel	| Linux alpha 4.19.0-10-amd64 #1 SMP Debian 4.19.132-1 (2020-07-24) x86_64 GNU/Linux
Architecture	| amd64 (2x Intel Xeon X5690), 288GB ECC DDR3
ZFS Version	| 0.7.12-2+deb10u2
SPL Version	| 0.7.12-2+deb10u1

### Describe the problem you're observing
ZFS appears to crash or hang on commiting TXGs during periods of heavy CPU or IO loads. On this particular machine, there are several virtual machines 
### Describe how to reproduce the problem
Heavy IO loads on the system will trigger the hang. The most recent incident happened when extracting and moving a 55gb file that was downloaded with nzbGet. I have also seen this happen when receiving large files with rsync over a gigabit connection, and also when a virtual machine on the server is performing a lot of I/O
### Include any warning/errors/backtraces from the system logs
Large, so instead of pasting it here I have attached it to the reply. 
[kern.log](https://github.com/openzfs/zfs/files/5196130/kern.log)

**Comment by pengc99** [comment]
```root@alpha:/var/log# zpool status
  pool: zfs_root
 state: ONLINE
  scan: resilvered 3.85T in 147h39m with 0 errors on Wed Sep  2 00:25:08 2020
config:

        NAME                                                       STATE     READ WRITE CKSUM
        zfs_root                                                   ONLINE       0     0     0
          raidz1-0                                                 ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy0-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy1-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy2-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy3-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy4-lun-0  ONLINE       0     0     0
          raidz1-1                                                 ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy5-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy6-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy7-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy8-lun-0  ONLINE       0     0     0
            pci-0000:01:00.0-sas-exp0x500065b36789abff-phy9-lun-0  ONLINE       0     0     0
        logs
          nvme0n1p1                                                ONLINE       0     0     0
        cache
          nvme0n1p2                                                ONLINE       0     0     0

errors: No known data errors
```

**Comment by gmelikov** [comment]
Duplicate of https://github.com/openzfs/zfs/issues/8673 , closed in https://github.com/openzfs/zfs/pull/10148 , released in 0.8.4, please use this version.

**Comment by pengc99** [comment]
Yup, I see that now. I'll upgrade to 0.8.4 with backports and see how it runs. Thanks for the quick update!
